Kontent.Ai.Delivery.Caching/DistributedCacheManager.cs

```diff
@@ -15,25 +16,38 @@ internal class DistributedCacheManager : IDeliveryCacheManager
     {
         private readonly IDistributedCache _distributedCache;
         private readonly DeliveryCacheOptions _cacheOptions;
+        private readonly ILogger _logger;
 
         /// <summary>
         /// Initializes a new instance of <see cref="DistributedCacheManager"/>
         /// </summary>
         /// <param name="distributedCache">An instance of an object that represent distributed cache</param>
         /// <param name="cacheOptions">The settings of the cache</param>
-        public DistributedCacheManager(IDistributedCache distributedCache, IOptions<DeliveryCacheOptions> cacheOptions)
+        /// <param name="loggerFactory">The factory used to create loggers.</param>
+        public DistributedCacheManager(IDistributedCache distributedCache,
+            IOptions<DeliveryCacheOptions> cacheOptions,
+            ILoggerFactory loggerFactory)
         {
             _distributedCache = distributedCache ?? throw new ArgumentNullException(nameof(distributedCache));
             _cacheOptions = cacheOptions.Value ?? new DeliveryCacheOptions();
+            _logger = loggerFactory.CreateLogger(nameof(DistributedCacheManager));
         }
 
         /// <inheritdoc />
         public async Task<T> GetOrAddAsync<T>(string key, Func<Task<T>> valueFactory, Func<T, bool> shouldCache = null, Func<T, IEnumerable<string>> dependenciesFactory = null) where T : class
         {
-            var (Success, Value) = await TryGetAsync<T>(key);
-            if (Success)
+            try
             {
-                return Value;
+                var (Success, Value) = await TryGetAsync<T>(key);
+                if (Success)
+                {
+                    return Value;
+                }
+            }
+            catch(Exception ex) when (ex is not ArgumentNullException && _cacheOptions.DistributedCacheResilientPolicy == DistributedCacheResilientPolicy.FallbackToApi) 
+            {
+                _logger.LogInformation(ex, "Distributed cache is not available. Default DeliveryClient was used to get content from Delivery API.");
```

Comment:
Is information the correct level?

Reply:
Honestly, I don't know how to determine the correct severity of this issue, so let's go with warning and wait for some feedback :)